dhis-2/dhis-api/src/main/java/org/hisp/dhis/scheduling/JobProgress.java

```diff
@@ -144,6 +138,18 @@ default boolean isSkipCurrentStage() {
     return isCancelled();
   }
 
+  /*
+  Error reporting API:
+  */
+
+  default void addError(ErrorCode code, String uid, String type, Integer index, String... args) {
+    addError(code, uid, type, index, List.of(args));
+  }
+
+  default void addError(ErrorCode code, String uid, String type, Integer index, List<String> args) {
+    // default implementation is a NOOP, we don't remember or handle the error
```

Comment:
I don't follow this reasoning for this empty method.
Do only certain jobs care about errors & they would then override this method? (which would result in the persisting of the error(s)?
And if jobs using this default method have errors, they are silently swallowed?

Reply:
This is not related to certain jobs but only one implementation of the `JobProgress` interfaces wants to record them and here we have no access to the implementation. To avoid having to implement the method in all other implementation with an empty method it is instead implemented like that as `default`.

Reply:
For the default to be an empty method suggests it shouldn't be part of the interface. It feels like we're adding this confusing behaviour to satisfy only 1 implementer.
Could a comment in code be added at least to explain why we don't want to handle errors here (handling errors seems like a very common use case) as the default and when would overriding be necessary. This would at least help someone reading the code to hopefully understand the reasoning for it & when it should be overridden.

Reply:
I'll do that in the next PR I was planning on doing anyhow